### DISCUSSION TITLE
Using Azure APIM as proxy for SQS SDK for .net - how to pass subscription key

### DISCUSSION THREAD

**Discussion by eilena1** [aws/aws-sdk-net#3188] [Q&A]
Hi, im trying to set up the SQS ClientConfig to use a Azure APIM API as a proxy to receive messages from an SQS queue. Im all new to AWS and SQS. My test works fine when i run it locally and directly agaist the queue. 

Ive set up the clientconfig:
```
var cred = new BasicAWSCredentials(secret1, secret2);
var sqsClientConfig = new AmazonSQSConfig()
{
    AuthenticationRegion = "eu-north-1",
    ProxyPort = 3443,
    ProxyHost = "http://xxxxxx/awssqsproxy/receivemessageproxy?subscription-key=xxxxxxx",
    ServiceURL = "https://sqs.eu-north-1.amazonaws.com/userid/queuename"
};
var sqsClient = new AmazonSQSClient(cred, sqsClientConfig);
```

Ive also tested with using the
```
sqsClientConfig.SetWebProxy(proxy);
```

I was hoping i could pass the subscription key as querystring instrad of a header.

I prefer getting this to work through the SDK. 
But I have tested out the backup plan with creating the request myself and pass the request though APIM to the sqs api. And that works. 

Any help is appreciated.

**Comment by dscpinheiro** [comment]
(I'm not familiar with Azure APIM, so apologies if I'm not understanding the use case)

In your example, isn't the subscription key already being passed as a query string?

**Reply by ashishdhingra** [reply]
@eilena1 Please review the last https://github.com/aws/aws-sdk-net/discussions/3188#discussioncomment-8529213.
